### DISCUSSION TITLE
Error when installing liquidsoap

### DISCUSSION THREAD

**Discussion by ByteEater-pl** [savonet/liquidsoap#1775] [Installation]
When I try to install liquidsoap (bulid from AUR – the only option showing in Pamac) on Linux 5.10.2-2-MANJARO #1 SMP PREEMPT, I get `Error: Library "ppxlib.metaquot" not found.`. I then asks for confirmation to proceed with 1 pending installation task and when I confirm, takes even longer that up to this point, installing multiple packages. I wasn't not sure if and how the resulting functionality of installed liquidsoap would be impaired.

But then I arrived at the apparently same point. So I let it continue again and saw the process starting over. Then I cancelled, having come to the realization that liquidsoap currently seems incompatible with Manjaro. Here's the error with some context:
```
==> Starting build()...
File "src/syntax/dune", line 8, characters 7-22:
8 |   (pps ppxlib.metaquot))
           ^^^^^^^^^^^^^^^
Error: Library "ppxlib.metaquot" not found.
Hint: try:
  dune external-lib-deps --missing @install
==> ERROR: A failure occurred in build().
    Aborting...
```
Moreover, issuing the command suggested `dune external-lib-deps --missing @install` causes only this to be displayed: `Done: 0/0 (jobs: 0)`.

**Comment by toots** [comment]
Hi! How are you building liquidsoap?

**Comment by ByteEater-pl** [comment]
With Pamac. It fetches liquidsoap 1.4.3-1 (currently) from https://aur.archlinux.org/packages/liquidsoap.

**Comment by toots** [comment]
It looks like the error is from the package itself. Have you tried installing via `opam`?

**Comment by ByteEater-pl** [comment]
No, at least not yet. Your question is noted (so I know that if I have further feedback after trying, it belongs here) and suggestion (if it's that; also present on the website) appreciated.

**Comment by ByteEater-pl** [comment]
@toots
> Have you tried installing via `opam`?

Yes, I issued the command `opam depext taglib mad lame cry samplerate liquidsoap`, but installation failed with this message:
```
# Detecting depexts using vars: arch=x86_64, os=linux, os-distribution=manjaro, os-family=arch
# The following system packages are needed:
libsamplerate
pcre
Distribution manjaro is not supported
```
The particularly odd thing is that both libsamplerate and pcre are present on my system.

**Comment by toots** [comment]
Looks like your distribution is not supported yet for external distribution. In this case, you should make sure that you have installed all the required packages and proceed directly with:
```
opam install taglib mad lame cry samplerate liquidsoap
```
